src/jdataview.js

```diff
@@ -1,833 +1,465 @@
-				buffer = createPixelData(buffer);
-			} else {
-				buffer = new Array(buffer);
-				for (var i = 0; i < buffer.length; i++) {
-					buffer[i] = 0;
-				}
-			}
-			return buffer;
-
-		case 'string':
-			buffer = getCharCodes(buffer);
-			/* falls through */
-		default:
-			if ('length' in buffer && !(
-				(NODE && compatibility.NodeBuffer && Buffer.isBuffer(buffer)) ||
-				(compatibility.ArrayBuffer && is(buffer, ArrayBuffer)) ||
-				(BROWSER && compatibility.PixelData && is(buffer, CanvasPixelArray))
-			)) {
-				if (NODE && compatibility.NodeBuffer) {
-					buffer = new Buffer(buffer);
-				} else
-				if (compatibility.ArrayBuffer) {
-					if (!is(buffer, ArrayBuffer)) {
-						buffer = new Uint8Array(buffer).buffer;
-						// bug in Node.js <= 0.8:
-						if (!is(buffer, ArrayBuffer)) {
-							buffer = new Uint8Array(arrayFrom(buffer, true)).buffer;
-						}
-					}
-				} else
-				if (BROWSER && compatibility.PixelData) {
-					buffer = createPixelData(buffer.length, buffer);
-				} else {
-					buffer = arrayFrom(buffer);
-				}
-			}
-			return buffer;
-	}
-};
-
-function pow2(n) {
-	return (n >= 0 && n < 31) ? (1 << n) : (pow2[n] || (pow2[n] = Math.pow(2, n)));
-}
-
-jDataView.is = function (view) {
-	return view && view.jDataView;
-};
-
-jDataView.from = function () {
-	return new jDataView(arguments);
-};
-
-function Uint64(lo, hi) {
-	this.lo = lo;
-	this.hi = hi;
-}
-
-jDataView.Uint64 = Uint64;
-
-Uint64.prototype.valueOf = function () {
-	return this.lo + pow2(32) * this.hi;
-};
-
-Uint64.fromNumber = function (number) {
-	var hi = Math.floor(number / pow2(32)),
-		lo = number - hi * pow2(32);
-
-	return new Uint64(lo, hi);
-};
-
-function Int64(lo, hi) {
-	Uint64.apply(this, arguments);
-}
-
-jDataView.Int64 = Int64;
+import { getCharCodes, wrapBuffer } from "./helpers";
+
+/**
+ * jDataView provides a layer on top of the built-in `DataView` with a plethora of utilities to make working with binary data a pleasure.
+ *
+ * Create a `jDataView` using either `new jDataView()` or `jDataView.from()`.
+ *
+ * [Read the docs](https://github.com/jDataView/jDataView/wiki)
+ */
+export class jDataView {
+	#bitOffset;
+	#bytePointer;
+
+	/**
+	 * jDataView provides a layer on top of the built-in `DataView` with a plethora of utilities to make working with binary data a pleasure.
+	 *
+	 * [Read the docs](https://github.com/jDataView/jDataView/wiki)
+	 */
+	constructor(buffer, byteOffset, byteLength, littleEndian) {
+		/**
+		 * The jDataView instance. Can be used to check if something really is a jDataView.
+		 */
+		this.jDataView = this;
+
+		if (buffer instanceof jDataView) {
+			const result = buffer.slice(byteOffset, byteOffset + byteLength);
+			result.littleEndian = littleEndian ?? result.littleEndian;
+			return result;
+		}
+
+		/**
+		 * The internal `ArrayBuffer` that jDataView is a view on
+		 */
+		this.buffer = wrapBuffer(buffer); // Convert strings, arrays, etc to `ArrayBuffer`s
```

Comment:
Now that we always wrap DataView, it seems we duplicate those 3 properties between our instance and the nested DataView.

Assuming we never modify them (and I suppose not, since we have separate `#bytePointer`), let's get them from the native instance when necessary instead to reduce duplication.

Since we still want to conform to DataView interface, it probably means declaring getters for these 3 properties that just return inner values instead.